llm_bioinformatics_research/src/Components/HomeScreen/HomeScreen.js

```diff
@@ -13,7 +14,36 @@ const CustomPaper = ({ children, title }) => (
   </Paper>
 );
 
-const HomeScreen = () => {
+function HomeScreen({ setIsLoggedIn }) {
+  const navigate = useNavigate();
+  const [openDialog, setOpenDialog] = React.useState(false);
+
+  useEffect(() => {
```

Comment:
One thing I noticed is that if I navigate from the home page to another page in the application and then back to the home page and then press the back arrow, the logout dialogue pops up. However, it should take me back to the page I was on before home. I think the logout dialogue should only pop up if the login page was the previously visited page. 

For instance, if my navigation sequence was like this:
Login --> Home --> API Documentation --> Home
and then I pressed the back arrow from Home, it should take me back to the API Documentation instead of showing a logout dialogue.